bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,432 @@
+  # for age = 4, include 4 and below
+  max_birthday_clause = "students.birthday <= DATE_ADD(NOW(), INTERVAL -#{age} YEAR) AND" unless age <= 4
+
+  sql = "
+    UPDATE contact_rollups, (
+    SELECT DISTINCT sections.user_id AS teacher_user_id
+    FROM #{DASHBOARD_DB_NAME}.users AS students
+      INNER JOIN #{DASHBOARD_DB_NAME}.followers AS followers
+        ON followers.student_user_id = students.id
+      INNER JOIN #{DASHBOARD_DB_NAME}.sections AS sections
+        ON sections.id = followers.section_id
+    WHERE
+      -- filter by student age
+      #{min_birthday_clause}
+      #{max_birthday_clause}
+      -- filter out teachers who have joined section
+      LENGTH(students.email) = 0
+    ) src
+    SET contact_rollups.ages_taught = CONCAT(COALESCE(CONCAT(contact_rollups.ages_taught, ','), ''), '#{age}')
+    WHERE contact_rollups.dashboard_user_id = teacher_user_id"
+
+  PEGASUS_DB.run sql
+  log_completion(start)
+end
+
+def update_ages_taught
+  # Computing all ages taught per teacher in one massive query was computationally
+  # overwhelming - the query would never complete. After experimenting with a variety
+  # of approaches, the most performant solution proved to be slicing first by student
+  # age (for each integer age) and calculating which teachers teach that age.
+  (4..21).each do |age|
+    update_ages_taught_for_age(age)
+  end
+end
+
+def update_district
+  start = Time.now
+  log "Updating district information from users"
+
+  # note that user district information seems less good than dashboard.pd_enrollments. So
+  # far no user has had district info where dashboard.pd_enrollments did not. And user district info
+  # has district names like "open csp".
+  districts = District.all.index_by(&:id)
+
+  users = User.where("length(email) > 0")
+
+  users.find_each do |user|
+    PEGASUS_DB[:contact_rollups].where(email: user.email).
+      update(school_name: user.ops_school) unless user.ops_school.nil?
+
+    unless user.district_id.nil?
+      district = districts[user.district_id]
+      unless district.nil?
+        PEGASUS_DB[:contact_rollups].where(email: user.email).update(district_name: district.name)
+      end
+    end
+  end
+  log_completion(start)
+
+  start = Time.now
+  log "Updating district information from dashboard.pd_enrollments"
+  DASHBOARD_REPORTING_DB_READONLY[:pd_enrollments].exclude(email: nil).exclude(school_info_id: nil).
+    select_append(:school_districts__name___district_name).select_append(:school_districts__updated_at___district_updated_at).
+    inner_join(:school_infos, :id => :school_info_id).
+    inner_join(:school_districts, :id => :school_district_id).order_by(:district_updated_at).each do |pd_enrollment|
+
+    PEGASUS_DB[:contact_rollups].where(email: pd_enrollment[:email]).update(district_name: pd_enrollment[:district_name], district_city: pd_enrollment[:city],
+      district_state: pd_enrollment[:state], district_zip: pd_enrollment[:zip])
```

Comment:
Not seeing that in commit history, although maybe I missed something. In general, it's been going the other direction. I started trying to use ActiveRecord/Sequel throughout but kept finding cases where writing raw SQL was 20x faster so was generally converting in that direction. Again, this part takes very little time to execute (I measure everything!) so I am not concerned about optimizing it.